### PR TITLE
8259709: Disable SHA-1 XML Signatures

### PR DESCRIPTION
**Pull request by seanjmullan**
Please review this change to disable XML signatures that use SHA-1 based digest or signature algorithms. SHA-1 is weak and is not a recommended algorithm for digital signatures. This will improve out of the box security by restricting XML signatures that use SHA-1 algorithms.

CSR: https://bugs.openjdk.java.net/browse/JDK-8261246
Release Note: https://bugs.openjdk.java.net/browse/JDK-8261364

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8259709](https://bugs.openjdk.java.net/browse/JDK-8259709): Disable SHA-1 XML Signatures

### Reviewers
 * [Rajan Halade](https://openjdk.java.net/census#rhalade) (@rhalade - **Reviewer**) ⚠️ Review applies to fd586fc77ece31d83f78163ae024fe91d638a7cf
 * [Weijun Wang](https://openjdk.java.net/census#weijun) (@wangweij - **Reviewer**) ⚠️ Review applies to fd586fc77ece31d83f78163ae024fe91d638a7cf

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2463/head:pull/2463`
`$ git checkout pull/2463`

### CODE REVIEW

**Review by rhalade — APPROVED** [review]

**Review by wangweij — APPROVED** [review]
Change looks good.

**Comment by wangweij** [comment]
All test changes are about re-enable disabled algorithms. Do we have a test on ensuring disabled algorithms are indeed disabled?

File: test/lib/jdk/test/lib/security/SecurityUtils.java [hunk 1]
```diff
@@ -70,5 +70,32 @@ private static void removeFromDisabledAlgs(String prop, List<String> algs) {
         Security.setProperty(prop, value);
     }
 
+    /**
+     * Removes the specified algorithms from the
+     * jdk.xml.dsig.secureValidationPolicy security property. Matches any
+     * part of the algorithm URI.
+     */
+    public static void removeAlgsFromDSigPolicy(List<String> algs) {
```
Comment (wangweij) [inline test/lib/jdk/test/lib/security/SecurityUtils.java:78]: How about use `String... algs` as arguments?

Comment (seanjmullan) [reply test/lib/jdk/test/lib/security/SecurityUtils.java:78]: Yes, that is nicer.

**Comment by seanjmullan** [comment]
> All test changes are about re-enabling disabled algorithms. Do we have a test on ensuring disabled algorithms are indeed disabled? How about we set "org.jcp.xml.dsig.secureValidation" to false everywhere in the existing tests and add a new dedicated test to check for disabled algorithms/key sizes etc.

That is what test/jdk/javax/xml/crypto/dsig/SecureValidationPolicy.java does, see this code block on lines 65-69:

```
        for (String alg : restrictedAlgs) {
            if (!Policy.restrictAlg(alg)) {
                throw new Exception(alg + " alg not restricted");
            }
        }

```

**Comment by wangweij** [comment]
> > All test changes are about re-enabling disabled algorithms. Do we have a test on ensuring disabled algorithms are indeed disabled? How about we set "org.jcp.xml.dsig.secureValidation" to false everywhere in the existing tests and add a new dedicated test to check for disabled algorithms/key sizes etc.
> 
> That is what test/jdk/javax/xml/crypto/dsig/SecureValidationPolicy.java does, see this code block on lines 65-69:
> 
> ```
>         for (String alg : restrictedAlgs) {
>             if (!Policy.restrictAlg(alg)) {
>                 throw new Exception(alg + " alg not restricted");
>             }
>         }
> ```

This is only about checking the parsing function of the Policy class. I would be more confident if an actual validation call is made.

I have a test on PSS at https://github.com/openjdk/jdk/pull/1206/files#diff-fa025c90d09b130fe3a8da89b4ebc5fc3ec157ded54b858dabe8241d383880ab. Maybe I can enhance it to contain more algorithms.

**Pull request closed without merging**